File: TransCoder_tokenized_test_set_functions/cpp/CHECK_ROWS_MATRIX_CIRCULAR_ROTATIONS.cpp
bool isPermutedMatrix ( int mat [ MAX ] [ MAX ], int n ) {
  string str_cat = "";
  for ( int i = 0;
  i < n;
  i ++ ) str_cat = str_cat + "-" + to_string ( mat [ 0 ] [ i ] );
  str_cat = str_cat + str_cat;
  for ( int i = 1;
  i < n;
  i ++ ) {
    string curr_str = "";
    for ( int j = 0;
    j < n;
    j ++ ) curr_str = curr_str + "-" + to_string ( mat [ i ] [ j ] );
    if ( str_cat . find ( curr_str ) == string :: npos ) return false;
  }
  return true;
}

File: TransCoder_tokenized_test_set_functions/python/CHECK_ROWS_MATRIX_CIRCULAR_ROTATIONS.py
def isPermutedMatrix ( mat, n ) :
    str_cat = ""
    for i in range ( n ) :
        str_cat = str_cat + "-" + str ( mat [ 0 ] [ i ] )
    str_cat = str_cat + str_cat
    for i in range ( 1, n ) :
        curr_str = ""
        for j in range ( n ) :
            curr_str = curr_str + "-" + str ( mat [ i ] [ j ] )
        if ( str_cat . find ( curr_str ) ) :
            return True
    return False
